src/classes/RD2_PauseForm_CTRL.cls

```diff
@@ -0,0 +1,270 @@
+
+        return isClosed;        
+    }
+
+    /**
+     * @description Returns the Paused Reason value and label select options
+     * @return PausedReason
+     */
+    private static PausedReason buildPausedReason() {
+        PausedReason reason = new PausedReason();
+        reason.label = System.Label.RD2_PausedReasonLabel;
+        reason.options = getPausedReasonOptions();
+        reason.placeholder = System.Label.lblSelectOptionPlaceholder;
+
+        return reason;
+    }
+
+    /**
+     * @description Returns the Paused Reason value and label select options
+     * @return void
+     */
+    private static void populateScheduleData(PauseData pause, Id rdId) {
+        RecurringDonationSchedule__c schedule = pauseHandler.getPauseSchedule(rdId);
+
+        if (schedule == null) {
+            return;
+        }
+
+        pause.pausedReason.value = schedule.StatusReason__c;
+        pause.scheduleId = schedule.Id;
+    }
+
+    /**
+     * @description Returns the Paused Reason value and label select options
+     * @return List<Option>
+     */
+    private static List<Option> getPausedReasonOptions() {
+        List<Option> options = new List<Option>();
+        List<Schema.PicklistEntry> entries = npe03__Recurring_Donation__c.ClosedReason__c.getDescribe().getPicklistValues();
+
+        for (Schema.PicklistEntry entry : entries) {
+            if (entry.isActive()) {                
+                options.add(
+                    new Option()
+                        .withValue(entry.getValue())
+                        .withLabel(entry.getLabel())
+                );
+            }
+        }
+        return options;
+    }
+    
+    /**
+     * @description Builds future projected installments for the Recurring Donation
+     * @param recordId Record Id
+     * @param maxRowDisplay Number of installments to display
+     * @return Map<String, Object>
+     */
+    @AuraEnabled
+    public static Map<String, Object> getInstallments(Id recordId, Integer maxRowDisplay) {
+        Map<String, Object> dataByKey = new Map<String, Object>();
+        try {
+            dataByKey = RD2_VisualizeScheduleController.getInstallments(recordId, maxRowDisplay);
+
+        } catch (Exception e) {
+            if (!e.getMessage().contains(System.Label.RD2_ScheduleVisualizerMessageNoSchedule)) {
+                throw e;
+            }
+        }
+
+        return dataByKey;
+    }
+
+    /**
+     * @description 
+     * @param 
+     * @return void
+     */
+    @AuraEnabled
+    public static void savePause(String jsonPauseData) {
+        try {
+            PauseData pause = (PauseData) JSON.deserialize(jsonPauseData, PauseData.class);
+
+            pauseHandler.cancelPauseSchedule(pause.rdId);
+
+            //create new pause schedule
+            RecurringDonationSchedule__c schedule = buildPauseSchedule(pause);
+            if (schedule.StartDate__c != null && schedule.EndDate__c != null) {
+                insert schedule;
```

Comment:
The part I'm a little unsure is whether a standard User has insert permissions into a protected object

Reply:
Thank you! The existing pause deactivation (update) and new pause insert have been moved into the schedule service.